### PR TITLE
Updated RESTful API tutorial to Eliom 6

### PR DESCRIPTION
**Pull request by colin-gd**
Migration to Eliom 6, using ppx_deriving_yojson instead of yojson-deriving, integrating Ocsipersist's updates

### CODE REVIEW

File: src/rest.wiki [hunk 1]
```diff
@@ -242,22 +240,26 @@ or a list of all existing locations otherwise.
...
 let read_handler id_opt () =
+  let%lwt db = db in
   match id_opt with
   | None ->
     Ocsipersist.fold_step
-      (fun id loc acc -> Lwt.return ((id, loc) :: acc)) db []
-    ~>>= fun locations ->
-    let json = Yojson.to_string<locations> locations in
+      (fun id loc acc -> Lwt.return ((id, loc) :: acc)) db []  
+    >>= fun locations ->
```
Comment (vasilisp) [inline src/rest.wiki]: We do need the `~`. It is used to escape our Wiki syntax, which assigns meaning to `>>`. (Same issue in multiple places.)

File: src/rest.wiki [hunk 2]
```diff
@@ -277,21 +280,27 @@ let edit_handler_aux ?(create = false) id_opt (content_type, raw_content_opt) =
...
-          Ocsipersist.add db id location ~>>= fun () ->
+           else
+            Ocsipersist.find db id >>= fun _ -> Lwt.return_unit)
+          >>= fun () ->
+          let location =
+            (let open Result in
+            let loc_result = location_of_yojson (Yojson.Safe.from_string location_str) in
+            (function
+              Ok loc-> loc
+            | Error _ -> raise Deriving_Yojson.Failed ) loc_result)  in
```
Comment (vasilisp) [inline src/rest.wiki]: You could just `match loc_result with ...` instead of introducing a function.

File: src/rest.wiki [hunk 3]
```diff
@@ -277,21 +280,27 @@ let edit_handler_aux ?(create = false) id_opt (content_type, raw_content_opt) =
...
+          Ocsipersist.add db  id location >>= fun () ->
           send_success ())
       (function
         | Not_found ->
           send_error ~code:404 ("Location not found: " ^ id)
         | Deriving_Yojson.Failed ->
-          send_error ~code:400 "Provided JSON is not valid")
+          send_error ~code:400 "Provided JSON is not valid"
+        | _ -> assert false)
```
Comment (vasilisp) [inline src/rest.wiki]: I am not sure what will happen here. Probably an internal server error? In any case, you can explicitly do something like `send_error ~code:500 "Internal server error"`

**Comment by colin-gd** [comment]
I've taken care of you remarks, let me know if you find something else.

File: src/rest.wiki [hunk 4]
```diff
@@ -259,7 +259,7 @@ let read_handler id_opt () =
         | Not_found ->
           (* [id] hasn't been found, return a "Not found" message *)
           send_error ~code:404 ("Resource not found: " ^ id)
-        | _ -> assert false)
+        | _ ->  send_error ~code:500 "Internal server error )
```
Comment (vasilisp) [inline src/rest.wiki]: Closing `"` :)

Comment (colin-gd) [inline src/rest.wiki]: Done!! Sorry about that :)

**Comment by vasilisp** [comment]
Merged, thank you!

**Pull request merged**